### PR TITLE
Add blocks and dynamic schedule page

### PR DESCRIPTION
**Pull request by evan10s**
Fancy things!  See the details below:

- Updated generic pages to be a one-column layout
- Added dynamic schedule page (generated using the Google Calendar API)
- Added "blocks" - reusable sections of information that can be included in special one and two column layouts
- Made the home.md file in /event_page appear when you go to /

### CODE REVIEW

**Comment by evan10s** [comment]
Closes #5, #9, #10, #11

**Review by mjkaufer — CHANGES_REQUESTED** [review]
I'm not a good person to get feedback for the Jekyll but w.r.t. to the JS, looks pretty good, small requests here, big request at the end:

* Get rid of console logs
* Add a comment around the anonymous function so ppl in the future editing this don't get shook

I'm pretty sure there's a really big XSS vulnerability via the calendar (i.e. event title called `<script>alert('lmao')</script>`). On one hand, that shouldn't be an issue because we control the calendar, but on the other I feel like we shouldn't have it as a potential issue regardless. I think you can escape the string, or since you're using jQuery, you can add your table elements to the DOM and then set their text. Also maybe google calendar already escapes stuff, but I'm not sure, and would rather not risk it.

Also, you included an API key in your `schedule.md` which implies we'd need to put it there consistently in the future, and biodomes mandates our repos are public. Would there be any way to make it an environment variable instead?

Otherwise, dope work!

**Comment by evan10s** [comment]
With the calendar API key, that's exposed to the client.  So I thought it's fine to keep in the repo.  We can restrict the key so it only works for certain http referers, like only subdomains of *.hack.gt

The other stuff should all be fixed, so I'll take a look at those items

**Comment by evan10s** [comment]
To do list of things to fix:

- [x] Remove all `console.log` statements in the JS
- [x] Add a comment about the anonymous function in `schedule.js`
- [x] Use jQuery's `.text()` method to safely add calendar events to the DOM
- [x] Add methods to gracefully catch failed AJAX requests and retry automatically

File: _includes/schedule.js [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+        var scheduleId = "#schedule-" + this.num;
+        $(scheduleId).prev().text("The schedule couldn't be retrieved.  Please check your internet connection."     );
+    }
+
+    // Show a warning in the console if endDateTime comes before startDateTime
+    if (new Date(endDateTime) - new Date(startDateTime) < 0) {
+        console.warn("schedule.js: No calendar events will be fetched!  The date of the last calendar event to include in the schedule (endDateTime) is earlier than the date of the first calendar event to include in the schedule.  The current start date is " + new Date(startDateTime) + ".  The current end date is " + new Date(endDateTime) + ".");
+    }
+
+    var url;
```
Comment (mjkaufer) [inline _includes/schedule.js]: Why do you have `url` and `calId` namespaced outside of the for loop? Can't you change line 111 to `var calId = calendars[i]` and line 113 to `var url = ...`?

File: _includes/schedule.js [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+                    prevCurrentDay = currentDay;
+                }
+
+                startTime = prettyTime(new Date(events[i].start.dateTime));
+                endTimeAsDate = new Date(events[i].end.dateTime)
+                endTime = prettyTime(endTimeAsDate);
+                location = events[i].location || "";
+                oldClass = (currentTime - endTimeAsDate > 0) ? ' class="old"' : ""; //event already ended
+                schedule.append("<tr" + oldClass + "><td></td><td></td><td></td><td></td></tr>");
+                // schedule.append("<tr" + oldClass + "><td>" + events[i].summary + "</td><td>" + startTime + "</td><td>" + endTime + "</td><td> " + location + "</td></tr>");
```
Comment (mjkaufer) [inline _includes/schedule.js]: drop these bad boys from the codebase <3

File: _includes/schedule.js [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+        var scheduleId = "#schedule-" + this.num;
+
+        if (result.items.length == 0) {
+            $(scheduleId).prev().text("No events found");
+        } else {
+            result.items.sort(function(a,b) {
+                return new Date(a.start.dateTime) - new Date(b.start.dateTime)
+                    || new Date(a.end.dateTime) - new Date(b.end.dateTime)
+                    || a.summary > b.summary;
+                //return 1;
```
Comment (mjkaufer) [inline _includes/schedule.js]: old commented code, pls remove

File: _includes/head.html [hunk 4]
```diff
@@ -17,4 +17,6 @@
         {% for script in page.js %}<script src="{{ script }}"></script>
         {% endfor %}
     {% endif %}
+
+    <script src="/assets/jquery-3.3.1.min.js"></script>
```
Comment (illegalprime) [inline _includes/head.html]: Please remove this.

File: _includes/schedule.js [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+                ];
+
+                $(scheduleId + ' > table > tbody > tr:last-child > td').each(function(index) {
+                    $(this).text(eventValues[index]);
+                });
+            }
+
+            $(scheduleId).prev().addClass("hidden");
+            $(scheduleId).removeClass("hidden");
+        } //END else
```
Comment (mjkaufer) [inline _includes/schedule.js]: If you don't want to put this whole thing in an else statement, you can just `return false` on line 25 with `result.items.length == 0`, and then just run the rest of the code. I'll leave this up to you – either way, remove `//END else` pls

**Review by illegalprime — CHANGES_REQUESTED** [review]
:mag:

File: _includes/head.html (hunk 4 above)
Comment (illegalprime) [reply _includes/head.html]: also optionally es6?

Comment (evan10s) [reply _includes/head.html]: The schedule uses jQuery to render itself for simplicity, but I can move the JS stuff to the footer to reduce the impact on pageload times.

We could do es6, but I'm pretty sure there are still devices with browsers that still don't support it, and adding a transpiler like Babel seems too complex for the minimal benefit (it's not like we need `async await` for any of this).

**Review by mjkaufer — CHANGES_REQUESTED** [review]

**Review by mjkaufer — DISMISSED** [review]

**Review by illegalprime — DISMISSED** [review]

**Comment by illegalprime** [comment]
Looks good, please remove the merge commit by rebasing and then merge.

**Comment by mjkaufer** [comment]
FYI I think you could've used `classList` without writing extra stuff https://caniuse.com/#search=classlist

Also, if you're gonna write fallback code that works on older platforms, feel free to just omit the original 'bleeding edge' code, since the old code will work on all platforms as it is.

**Comment by evan10s** [comment]
Ah, ok.  That was from youmightnotneedjquery.com but I can fix that

**Review by mjkaufer — APPROVED** [review]

**Pull request merged**